src/main/java/com/socialmediaraiser/twitterbot/impl/personalAnalyzer/TweetInteraction.java

```diff
@@ -0,0 +1,59 @@
+package com.socialmediaraiser.twitterbot.impl.personalAnalyzer;
+
+import java.util.HashSet;
+import java.util.Set;
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.With;
+import lombok.val;
+
+@Getter
+@Builder
+@NoArgsConstructor
+@AllArgsConstructor
+public class TweetInteraction {
+
+  @With
+  private Set<String> answererIds  = new HashSet<>();
+  @With
+  private Set<String> retweeterIds = new HashSet<>();
+  @With
+  private Set<String> likersIds    = new HashSet<>();
+
+  // Using VAVR collections everywhere would be much less noisy
+  public TweetInteraction addAnswerer(String answerer) {
```

Comment:
I don't understand the purpose of this function ? Can't I just do a `getAnswererIds().add(answerer)` ?

Reply:
No, because you want to return a new `TweetInteraction` with an updated `answerersId`.
You never return a `void` nor change the current data when you are doing functional style programming.